linera-base/src/crypto.rs

```diff
@@ -424,6 +424,15 @@ impl CryptoHash {
         CryptoHash(hasher.finalize())
     }
 
+    /// Computes the hash of the given bytes.
+    pub fn new_from_bytes(bytes: &[u8]) -> Self {
+        use sha3::digest::Digest;
+
+        let mut hasher = sha3::Sha3_256::default();
+        bytes.write(&mut hasher);
+        CryptoHash(hasher.finalize())
+    }
```

Comment:
We already have
```rust
impl<Hasher> Hashable<Hasher> for [u8]
where
    Hasher: io::Write,
{
    fn write(&self, hasher: &mut Hasher) {
        hasher.write_all(self).expect("Hasher should not fail");
    }
}
```

which I would expect can be used for `Blob` 🤔

Reply:
Isn't that exactly what I'm using in line 432?

Reply:
Hmm, I guess what I meant is that we should be able to use `CryptoHash::new(blob)` w/o the new function but OTOH the argument would be a `Blob` type...

Maybe `impl AsRef<[u8]> for Blob` and call `as_ref` on a `Blob` before passing into `Cryptohash::new`?

I'm trying to check if we really need this new method.

Reply:
I think the idea with requiring `BcsHashable` for `new` is that we never accidentally hash something that we're not meant to (but that happens to be serializable).

That's why it's not implemented for `[u8]`. So I thought making `new_from_bytes` explicit would keep that, but allow us to simply take the SHA-3 hash for the blob bytes anyway.

Reply:
Where's the domain seed in the current (`new_from_bytes`) approach?

Reply:
Yes, that's what he means. I'll close this PR and find another way to avoid the cloning.